src/encoded/static/libs/ui/node_graph.js

```diff
@@ -0,0 +1,254 @@
+const circlePlus = '\u2295';
+const circleMinus = '\u2296';
+
+const slowAnimation = 400;
+const fastAnimation = 0;
+
+const mobileLimit = 400;
+
+let offset = -40;
+let doubleOffset = -60;
+
+const ellipseSettings = [
+    { cx: 2, cy: 2, rx: 12, ry: 9 },
+    { cx: 2, cy: 2, rx: 6.5, ry: 6 },
+    { cx: -5, cy: 10, rx: 12, ry: 9 },
+    { cx: -5, cy: 10, rx: 6.5, ry: 5 },
+    { cx: 10, cy: 10, rx: 12, ry: 9 },
+    { cx: 10, cy: 10, rx: 6.5, ry: 5 },
+];
+
+// Standardizes node names
+function nodeKeyName(name) {
+    return name.replace(/\s/g, '').toLowerCase();
+}
+
+// Creates a curved (diagonal) path from parent to the child nodes
+function diagonal(d, s) {
+    const path = `M${d.x},${d.y}C${d.x},${(d.y + s.y) / 2} ${s.x},${(d.y + s.y) / 2} ${s.x},${s.y}`;
+    return path;
+}
+
+const drawTree = (d3, targetDiv, data, fullWidth, fullHeight, margin, selectedNodes, setSelectedNodes) => {
+    let textWrapWidth = 115;
+    let characterLimitForWrap = 18;
+    const isMobile = fullWidth < mobileLimit;
+    if (isMobile) {
+        textWrapWidth = 80;
+        characterLimitForWrap = 15;
+        margin.top = 50;
+        margin.bottom = 80;
+        margin.left = 3;
+        margin.right = 3;
+        offset = -35;
+        doubleOffset = -65;
+    }
+
+    const width = fullWidth - margin.left - margin.right;
+    const height = fullHeight - margin.top - margin.bottom;
+
+    d3.select(targetDiv).select('svg').remove();
+
+    let internalSelectedNodes = selectedNodes;
+
+    const svg = d3.select(targetDiv).append('svg')
+        .attr('width', fullWidth)
+        .attr('height', fullHeight)
+        .attr('viewBox', `0 0 ${fullWidth} ${fullHeight}`)
+        .attr('preserveAspectRatio', 'xMidYMin meet')
+        .append('g')
+        .attr('transform', `translate(${margin.left},${margin.top})`);
+
+    const treemap = d3.tree().size([width, height]);
+    const root = d3.hierarchy(data, (d) => d.children);
+    root.x0 = width / 2;
```

Comment:
it may be nice to comment why this is done; for further reader. '2' is a magic number and division by 2 is a magic operation.